### ISSUE TITLE
No internet - No icons in Integrations & Devices pages

### ISSUE THREAD

**Issue by AlainPeccable** [home-assistant/frontend#17560]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

We had a large outage from the ISP provider for more than 10h. During that time the **Integrations & Devices pages** within the **configuration page** were **not showing any icons**.
Once internet returned all icons were back.

Note that all dashboards / entities & areas were displaying the icons. Everything was also working as expected while being offline.

### Describe the behavior you expected

Icons visible even when internet is lost.

### Steps to reproduce the issue

1. ISP off
2. Check integrations & Devices page
3.
...

### What version of Home Assistant Core has the issue?

2023.8.1

### In which browser are you experiencing the issue with?

Google Chrome 115.0.5790.171

### Which operating system are you using to run this browser?

Microsoft Windows [version 10.0.22000.2295

### Additional information

During ISP Outage:
![HA no web no icon](https://github.com/home-assistant/frontend/assets/132245586/5dbdc9ee-c63c-47a8-99b8-63d0b1211df8)

After ISP Outage:
![image](https://github.com/home-assistant/frontend/assets/132245586/e4771631-7e3c-41d1-9c19-c54715271b5a)

**Comment by steverep** [comment]
What you are describing happens because the device/integration logos are externally hosted and pulled from https://brands.home-assistant.io/.  Other Material Design icons are stored locally and cached with the rest of the app on your browser.  See related issue #15440 for more explanation.

However, it looks like we don't currently have any logic to properly cache those brand images, which should be fixed. 👍🏻

**Comment by steverep** [comment]
I've been working on caching the brand images in #17840, but currently running into some technical snags.  Hopefully it'll be worked out soon.

Also, please note that the frontend is only able to address this via the service worker, which won't apply to legacy browsers or non-https access.  For that, it must be handled at the origin server, so I opened https://github.com/home-assistant/brands/issues/4718.

**Comment by cyberhuman** [comment]
What is the reason the icons and logos are not stored locally instead of hosting them on an external website?
I'm writing a (private) custom integration and I'm quite surprised that my device shows a broken image icon without the ability to customize it.

**Comment by steverep** [comment]
> What is the reason the icons and logos are hosted on an external website instead of storing them locally? 

My understanding is that the main issue for the decision was size, but that question, while related, is off-topic and would be best discussed on the community forum or Discord chat.

> I'm writing a (private) custom integration and I'm quite surprised that my device shows a broken image icon without the ability to customize it.

You mean it shows a [placeholder image](https://brands.home-assistant.io/_/_placeholder/icon.png), or an actual error?  The latter would be a new issue in the frontend while the former is by design.


I know custom integrations are free to submit images to the brands website as well, but honestly I don't know if there's a way to add them locally only for private use.  That would be a feature request for core though, not the frontend.

**Comment by cyberhuman** [comment]
It shows an actual error loading the image:
![image](https://github.com/home-assistant/frontend/assets/979624/adc39150-e0ff-4585-970a-5abb352af1f4)
I noticed it's because the URL is https://brands.home-assistant.io/whatever/dark_icon.png but it should be https://brands.home-assistant.io/_/whatever/dark_icon.png for the placeholder image to work.

**Comment by cyberhuman** [comment]
It's a bit better with the placeholder image, but not that much of a difference.
![image](https://github.com/home-assistant/frontend/assets/979624/1bd21f0f-d5c9-4945-887a-0c825d569739)

**Comment by steverep** [comment]
Yeah that's a bug.  It should be either no image or the placeholder.  Can you please open another issue for that?

**Comment by steverep** [comment]
Just a quick status update... waiting on https://github.com/home-assistant/brands/pull/4722 before the frontend change can be merged.
